go/otgrpc/shared.go

```diff
@@ -18,6 +20,7 @@ type metadataReaderWriter struct {
 }
 
 func (w metadataReaderWriter) Set(key, val string) {
+	key = strings.ToLower(key)
```

Comment:
@shouichi 
1. I now agree that this line is probably a good approach, but that's because GRPC (or, really, the HPACK impl) *is* case-sensitive and requires lowercase.
2. Along those lines, the opentracing-go repo should not be changed IMO. For the true HTTP use case there is no need to adjust casing one way or the other.
3. There is "another way", which would be to use a *single* key in the GRPC metadata and explicitly pack/unpack the contents of the TextMap into a string. That's not "hard", but is annoying as it would have to be done in every GRPC-OT language (and done with perfect consistency, including all annoying escaping and so forth).

TL;DR let's keep this code just as you have it, but let's *also* add a comment explaining that the GRPC HPACK impl requires lowercase keys.

Thanks!